vernac/vernacinterp.ml

```diff
@@ -206,13 +206,13 @@ and vernac_load ~verbosely fname =
     CErrors.user_err Pp.(str "Files processed by Load cannot leave open proofs.");
   stack, pm
 
-and interp_control ~st ({ CAst.v = cmd } as vernac) =
+and interp_control ~st ({ CAst.v = cmd; loc } as vernac) =
   let time_header = mk_time_header vernac in
   List.fold_right (fun flag fn -> interp_control_flag ~time_header flag fn)
     cmd.control
     (fun ~st ->
        let before_univs = Global.universes () in
-       let pstack, pm = interp_expr ~atts:cmd.attrs ~st cmd.expr in
+       let pstack, pm = interp_expr ~loc:(Option.get loc) ~atts:cmd.attrs ~st cmd.expr in
```

Comment:
Option.get should be removed here, it is possible that a command has no location in the current API.

Reply:
I don't know how to express it with the CAst.t datatype, since the loc is an option, but the root AST node always has one, since it is the parser of Coq which parses it (even if the command was added with EXTEND) and our parser puts the loc.
Indeed CI is not failing.

I can use `CErrors.anomaly` if you prefer, but I really want the client code to receive a loc, not an loc option.

Reply:
It is more tricky than that, for example some plugins or extern do create AST nodes without location and these should be reentrant here. The API should remain well-typed instead of an assert, so indeed `interp_expr` needs to take the right type, if you use `vernac_expr located` the type will be already an option.